app/controllers/candidate_interface/continuous_applications/course_choices/provider_selection_controller.rb

```diff
@@ -5,9 +5,17 @@ class ProviderSelectionController < BaseController
       private
 
         def step_params
+          return provider_params if params[:provider_id].present?
```

Comment:
I don't mind present? for the present case first or blank for the blank case. 
The other cases won't apply, because `params[current_step]` will always be blank for the new action (but not for create).

Reply:
I think present? (the current code) is the ideal because the first condition will only run at new action (and it is only applied for the ones that comes back one page) and create will persist the params[:current_step].

Reply:
I copied the wrong code, sorry. I'll correct it in the next one.

Reply:
Not sure if you care about having just the `provider_id` in the params, but if you don't, I think you can do `return default_params if params[:provider_id].present?` or just a one liner with ternary `params[:provider_id].present? ? default_params : params`

Reply:
Done!